test_src/cross_field_diffusion_tests.cpp

```diff
@@ -8,6 +8,31 @@
 #include "curandInitialize.h"
 #include <thrust/execution_policy.h>
 
+template <typename T=double>
+bool compareVectors(std::vector<T> a, std::vector<T> b, T epsilon, T margin)
```

Comment:
I would like to move this functionality into test_utils.cpp at some point - this will be very useful for other tests as well!